apps/examples/avs_test/avs_test_main.cpp

```diff
@@ -0,0 +1,136 @@
+ * examples/avs_test/avs_test_main.cpp
+ *
+ *   Copyright (C) 2008, 2011-2012 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/init.h>
+#include <apps/platform/cxxinitialize.h>
+
+#include <fcntl.h>
+#include <wifi_manager/wifi_manager.h>
+
+#include <iostream>
+
+#include <avs-device-sdk/avs-device-sdk.h>
+
+#define SAY_OK_OR_ERROR(condition) if (ret == (condition)) { printf("Ok\n"); } else { printf("Error\n"); }
+
+extern "C" {
+
+static void wifi_sta_connected(wifi_manager_result_e result) {
+	printf("%s\n", __FUNCTION__);
+}
+
+static void wifi_sta_disconnected() {
+	printf("%s\n", __FUNCTION__);
+}
+
+static wifi_manager_cb_s wifi_callbacks = {
+	wifi_sta_connected,
+	wifi_sta_disconnected,
+	NULL,
+	NULL,
+	NULL,
+};
+
+int avs_test_main(int argc, char **argv)
+{
+#ifdef CONFIG_EXAMPLES_AVS_TEST_CXXINITIALIZE
+	up_cxxinitialize();
+#endif	
+	
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	wifi_manager_ap_config_s config;
+
+	/**
+	 * Need to sleep for WiFi initialized.
+	 */
+	int i = 0;
+	for (i = 0; i < 3; i++) {
+		printf("Wait (%d/3) sec...\n", i + 1);
+		sleep(1);
+	}	
+
+	printf("wifi_manager_init()...\n");
+	ret = wifi_manager_init(&wifi_callbacks);
+	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+
+	config.ssid_length = strlen(CONFIG_EXAMPLES_AVS_TEST_SSID);
```

Comment:
If we use config value directly, we should add some defines.
#ifndef CONFIG_XXXX
#define CONFIG_XXXX
#endif

Reply:
This is fixed.